File: level1/모의고사.py
def solution(answers):
    answer = []
    
    person = [[1,2,3,4,5],[2,1,2,3,2,4,2,5],[3,3,1,1,2,2,4,4,5,5]]
    
    max_c = -1
    for i in range(3):
        count = 0
        for j in range(len(answers)):
            if answers[j] == person[i][j%len(person[i])]:
                count += 1
        if max_c < count:
            answer = []
            answer.append(i+1)
            max_c = count
        elif max_c == count:
            answer.append(i+1)
            
    return answer

if __name__ == '__main__':
    answers = input()
    
    print(solution(answers))

File: level1/모의고사.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> answers) {    
    int student1[5] = {1,2,3,4,5};
    int student2[8] = {2,1,2,3,2,4,2,5};
    int student3[10] = {3,3,1,1,2,2,4,4,5,5};
    
    vector<int> correct(3,0);
    int a_size = answers.size();
    for(int i = 0; i < a_size; i++){
        if(student1[i%5] == answers[i]){
            correct[0]++;
        }
        if(student2[i%8] == answers[i]){
            correct[1]++;
        }
        if(student3[i%10] == answers[i]){
            correct[2]++;
        }
    }
    
    int s_max = -1;
    vector<int> answer;
    for(int i = 0; i < 3; i++){
        if(s_max < correct[i]){
            s_max = correct[i];
            vector<int> tmp;
            tmp.push_back(i+1);
            answer = tmp;
        }
        else if(s_max == correct[i]){
            answer.push_back(i+1);
        }
    }
    
    return answer;
}
